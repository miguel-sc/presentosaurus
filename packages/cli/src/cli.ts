#!/usr/bin/env node

import yargs from "yargs";

yargs
  .scriptName("pyro")
  .command({
    command: "build <slides>",
    describe: "Build HTML from markdown",
    builder: yargs =>
      yargs.positional("slides", {
        type: "string",
        default: "index.md",
        describe: "path to markdown file"
      }),
    handler: argv => {
      console.log(`building ${argv.slides}`);
    }
  })
  .help().argv;
