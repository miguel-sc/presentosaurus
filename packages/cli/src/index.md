test: test

---

# Sample Presentation

---

## Title

- meow
- moo

---

quack lol
