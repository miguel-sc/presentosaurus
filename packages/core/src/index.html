<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0"
    />
    <title>ðŸ”¥Pyro</title>
    <link rel="stylesheet" href="/build/pyro.css" />
    <script type="module" src="/build/pyro.esm.js"></script>
    <script nomodule src="/build/pyro.js"></script>
    <style>
      body {
        width: 100vw;
        height: 100vh;
        display: flex;
        align-items: center;
        justify-content: center;
        background-color: #f7f7f7;
      }
      pyro-presentation {
        width: 960px;
        height: 600px;
        box-shadow: 2px 2px 5px 0px rgba(0, 0, 0, 0.75);
      }
    </style>
  </head>
  <body>
    <pyro-presentation
      data-theme="dark"
      options='{"code": {"lineNumbers": true}}'
    >
      <pyro-slide>
        <pyro-code src="/assets/fizzbuzz.ts"></pyro-code>
      </pyro-slide>
      <pyro-slide>
        <pyro-math
          expression="f(x) = \int_{-\infty}^\infty\hat f(\xi)\,e^{2 \pi i \xi x}\,d\xi"
        ></pyro-math>
        <pyro-math
          expression="x = (-b +- sqrt(b^2-4ac)) / (2a)"
          ascii-math
        ></pyro-math>
        <pyro-math
          expression="\ce{Hg^2+ ->[I-] HgI2 ->[I-] [Hg^{II}I4]^2-}"
        ></pyro-math>
      </pyro-slide>
      <pyro-slide>
        <blockquote cite="https://www.huxley.net/bnw/four.html">
          <p>
            Words can be like X-rays, if you use them properlyâ€”theyâ€™ll go
            through anything. You read and youâ€™re pierced.
          </p>
          <footer>Aldous Huxley, <cite>Brave New World</cite></footer>
        </blockquote>
      </pyro-slide>
    </pyro-presentation>
  </body>
</html>
