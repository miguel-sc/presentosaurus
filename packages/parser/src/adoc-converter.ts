import asciidoctor from "@asciidoctor/core";

const htmlConverter = asciidoctor().Html5Converter.$new();

export const adocConverter = options => ({
  convert: (node, transform) => {
    switch (node.getNodeName()) {
      case "document":
        return `<pyro-presentation options='${JSON.stringify(
          options
        )}'>${node.getContent()}</pyro-presentation>`;
      case "section":
        return `<pyro-slide><h2>${node.getTitle()}</h2>${node.getContent()}</pyro-slide>`;
      case "listing":
        return `<pre><pyro-code language="${node.getAttribute(
          "language"
        )}">${node.getContent()}</pyro-code></pre>`;
      case "stem":
        return `<pyro-math ascii-math=${node.getAttribute("style") ===
          "asciimath"} expression="${node.getContent()}"></pyro-math>`;
      case "inline_quoted":
        if (node.type === "asciimath" || node.type === "latexmath") {
          return `<pyro-math inline ascii-math=${node.type ===
            "asciimath"} expression="${node.text}"></pyro-math>`;
        } else {
          return htmlConverter.convert(node, transform);
        }
      default:
        return htmlConverter.convert(node, transform);
    }
  }
});
