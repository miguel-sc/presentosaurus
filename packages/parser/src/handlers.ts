import detab from "detab";
import u from "unist-builder";

const code = (h, node) => {
  const value = node.value ? detab(node.value + "\n") : "";
  const language = node.lang && node.lang.match(/^[^ \t]+(?=[ \t]|$)/);
  const props = {
    language
  };

  return h(node.position, "pre", [
    h(node, "ps-code", props, [u("text", value)])
  ]);
};

const math = (h, node) => {
  const value = node.value ? detab(node.value) : "";
  const props = {
    expression: value
  };

  return h(node.position, "ps-math", props);
};

const inlineMath = (h, node) => {
  const value = node.value ? detab(node.value) : "";
  const props = {
    expression: value,
    inline: true
  };

  return h(node.position, "ps-math", props);
};

export const handlers = {
  code,
  math,
  inlineMath
};
