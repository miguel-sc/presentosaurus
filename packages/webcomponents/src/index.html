<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0"
    />
    <title>ðŸ”¥Pyro</title>
    <link rel="stylesheet" href="/build/pyro.css" />
    <script type="module" src="/build/pyro.esm.js"></script>
    <script nomodule src="/build/pyro.js"></script>
    <style>
      body {
        width: 100vw;
        height: 100vh;
        display: flex;
        align-items: center;
        justify-content: center;
        background-color: #f7f7f7;
      }
      pyro-presentation {
        width: 960px;
        height: 600px;
        box-shadow: 2px 2px 5px 0px rgba(0, 0, 0, 0.75);
      }
    </style>
  </head>
  <body>
    <pyro-presentation
      data-theme="dark"
      options='{"code": {"lineNumbers": true}}'
    >
      <pyro-slide slide-title="Syntax highlighting">
        <pyro-code src="/assets/fizzbuzz.ts"></pyro-code>
      </pyro-slide>
      <pyro-slide slide-title="Math & chemistry">
        <pyro-math
          expression="f(x) = \int_{-\infty}^\infty\hat f(\xi)\,e^{2 \pi i \xi x}\,d\xi"
        ></pyro-math>
        <pyro-math
          expression="x = (-b +- sqrt(b^2-4ac)) / (2a)"
          ascii-math
        ></pyro-math>
        <pyro-math
          expression="\ce{Hg^2+ ->[I-] HgI2 ->[I-] [Hg^{II}I4]^2-}"
        ></pyro-math>
      </pyro-slide>
      <pyro-slide>
        <blockquote cite="https://www.huxley.net/bnw/four.html">
          <p>
            Words can be like X-rays, if you use them properlyâ€”theyâ€™ll go
            through anything. You read and youâ€™re pierced.
          </p>
          <footer>Aldous Huxley, <cite>Brave New World</cite></footer>
        </blockquote>
      </pyro-slide>
      <pyro-slide
        background-image="https://media.giphy.com/media/5nsiFjdgylfK3csZ5T/giphy.gif"
      >
      </pyro-slide>
      <pyro-slide background-iframe="https://www.youtube.com/embed/dQw4w9WgXcQ">
      </pyro-slide>
    </pyro-presentation>
  </body>
</html>
