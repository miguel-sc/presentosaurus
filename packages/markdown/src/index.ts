import unified from "unified";
import markdown from "remark-parse";
import math from "remark-math";
import remark2rehype from "remark-rehype";
import { slides } from "./rehype-slides";
import format from "rehype-format";
import html from "rehype-stringify";
import YAML from "yaml";
import merge from "deepmerge";
import { handlers } from "./handlers";

export const mdToHtml = (md: string, options?) => {
  const [yamlOptions, ...mdslides] = md.split("\n---\n");

  const presentationOptions = merge(
    options || {},
    YAML.parse(yamlOptions) || {}
  );

  const converter = unified()
    .use(markdown)
    .use(math)
    .use(remark2rehype, { handlers })
    .use(slides, presentationOptions)
    .use(format)
    .use(html);

  return String(converter.processSync(mdslides.join("\n---\n")));
};
