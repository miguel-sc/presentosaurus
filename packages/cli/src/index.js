import { mdToHtml } from "@pyroslides/markdown";
import { defineCustomElements } from "@pyroslides/webcomponents/dist/esm/loader";
import "@pyroslides/webcomponents/dist/pyro/pyro.css";

defineCustomElements(window);

import mdslides from "./index.md";

document.body.innerHTML = mdToHtml(mdslides);
