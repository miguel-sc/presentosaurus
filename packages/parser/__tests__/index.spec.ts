import { mdToHtml, adocToHtml } from "../src";
import { readdirSync, readFileSync } from "fs";
import { join } from "path";

const fixturesPath = join(__dirname, "__fixtures__");
const mdslidesPath = join(fixturesPath, "markdown");
const mdslides = readdirSync(mdslidesPath);
const adocslidesPath = join(fixturesPath, "asciidoctor");
const adocslides = readdirSync(adocslidesPath);

test.each(mdslides)("should render snapshot of %s", slide => {
  const md = readFileSync(join(mdslidesPath, slide), {
    encoding: "utf8"
  });
  expect(mdToHtml(md)).toMatchSnapshot();
});

test.each(adocslides)("should render snapshot of %s", slide => {
  const md = readFileSync(join(adocslidesPath, slide), {
    encoding: "utf8"
  });
  expect(adocToHtml(md)).toMatchSnapshot();
});

test("should render document", () => {
  expect(mdToHtml("").startsWith("<!doctype html>")).toBe(false);
  expect(mdToHtml("", undefined, true).startsWith("<!doctype html>")).toBe(
    true
  );
});

test("should use document options", () => {
  expect(mdToHtml("", { document: { title: "titlestring" } }, true)).toMatch(
    "<title>titlestring</title>"
  );
});
