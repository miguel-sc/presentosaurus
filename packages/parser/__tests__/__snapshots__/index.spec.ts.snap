// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render snapshot of empty.adoc 1`] = `
"
<pyro-presentation options=\\"{}\\"></pyro-presentation>
"
`;

exports[`should render snapshot of empty.md 1`] = `
"
<pyro-presentation options=\\"{}\\">
  <pyro-slide></pyro-slide>
</pyro-presentation>
"
`;

exports[`should render snapshot of sample.adoc 1`] = `
"
<pyro-presentation options=\\"{}\\">
  <pyro-slide>
    <h2>Title Slide</h2>
    <div class=\\"paragraph\\">
      <p>Wuff</p>
    </div>
  </pyro-slide>
  <pyro-slide>
    <h2>Slide One</h2>
    <div class=\\"ulist\\">
      <ul>
        <li>
          <p>Foo</p>
        </li>
        <li>
          <p>Bar</p>
        </li>
        <li>
          <p>World</p>
        </li>
      </ul>
    </div>
  </pyro-slide>
  <pyro-slide>
    <h2>Slide Two</h2>
    <div class=\\"paragraph\\">
      <p>A Great Story</p>
    </div>
    <pyro-math ascii-math=\\"true\\" expression=\\"sqrt(4) = 2\\"></pyro-math>
  </pyro-slide>
  <pyro-slide>
    <h2>Slide Three</h2>
    <pre><pyro-code language=\\"js\\">const add = (a, b) => {
  return a + b;
};</pyro-code></pre>
  </pyro-slide>
</pyro-presentation>
"
`;

exports[`should render snapshot of sample.md 1`] = `
"
<pyro-presentation options=\\"{&#x22;code&#x22;:{&#x22;lineNumbers&#x22;:true},&#x22;math&#x22;:{&#x22;asciiMath&#x22;:true}}\\">
  <pyro-slide>
    <h1>Sample Presentation</h1>
  </pyro-slide>
  <pyro-slide>
    <h2>Title</h2>
    <ul>
      <li>meow</li>
      <li>moo</li>
    </ul>
  </pyro-slide>
  <pyro-slide>
    <p>
      <pyro-math expression=\\"\\\\frac{1}{2}\\" inline></pyro-math>
    </p>
    <pyro-math expression=\\"\\\\int f\\"></pyro-math>
  </pyro-slide>
  <pyro-slide>
    <pre><pyro-code language=\\"js\\">const add = (a, b) => {
  return a + b;
};
</pyro-code></pre>
  </pyro-slide>
</pyro-presentation>
"
`;
